Source/path.cpp

```diff
@@ -56,6 +57,36 @@ const Displacement PathDirs[8] = {
  */
 int8_t path_directions[9] = { 5, 1, 6, 2, 0, 3, 8, 4, 7 };
 
+bool IsTileNotSolid(Point position)
+{
+	return !nSolidTable[dPiece[position.x][position.y]];
+}
+
+bool IsTileSolid(Point position)
+{
+	if (position.x < 0 || position.y < 0 || position.x >= MAXDUNX || position.y >= MAXDUNY) {
+		return false;
+	}
+
+	return nSolidTable[dPiece[position.x][position.y]];
+}
+
+/**
+ * @brief Checks the position is solid or blocked by an object
+ */
+bool IsTileWalkable(Point position, bool ignoreDoors)
+{
+	if (dObject[position.x][position.y] != 0) {
+		int oi = dObject[position.x][position.y] > 0 ? dObject[position.x][position.y] - 1 : -(dObject[position.x][position.y] + 1);

```

Comment:
```suggestion
		int oi = abs(dObject[position.x][position.y]) - 1;
```
Saves having to index in to the array multiple times.

Reply:
Wish some one had pointed this out earlier, we should be using abs() every where for this :D